### DISCUSSION TITLE
5.1.0版本dledger集群时，定时消息不生效

### DISCUSSION THREAD

**Discussion by kejiancai** [apache/rocketmq#6617] [Q&A]
关于5.1.0版本，在单机或者普通主从集群情况下，定时可用，当部署为dledger集群时，定时只能对立即发送的数据有效，如果定时时间非立即执行，返回状态成功，并返回msgId，但是实际查询不到该msgId数据。不知道是否有人碰到过此情况，如果是我操作问题，希望帮忙指出，如果是该版本自身问题，希望此问题有助于提供帮助。

　　举例：如当前时间为2023-04-19 11:37:00，需要发送一条定时时间在2023-04-19 12:00:00的消息(该时间转换为1681876800)，setDeliverTimeMs(1681876800)设置之后，SendResult能够返回SendStatus.SEND_OK，且SendStatus.getMsgId()也是有值的，但是在集群内，在对应的tpic内无法查找到，通过message id也搜索不到（无论当前时间是否已经到了定时时间）。

**Comment by zhiliatom** [comment]
setDeliverTimeMs(1681876800)这个api填毫秒数，不是时间戳

**Reply by kejiancai** [reply]
使用开源客户端，setDeliverTimeMs这个是填写的时间戳哦

**Reply by zhiliatom** [reply]
不好意思看错了，message.setDelayTimeMs(10);是填间隔时间。有一种可能是客户端时间和服务端不一致，实际还未出时间轮

**Reply by kejiancai** [reply]
应该不是，我测试的时候，只加了10秒上去，而且不管是否出时间轮，在消息的MESSAGE ID那边是肯定要能查询到的，关键是现在没查询到，topic也没有这条数据，之前没有用dledger集群部署的时候，定时消息也都是正常的，用了dledger才出现了这个问题

**Comment by RongtongJin** [comment]
有可能与该修复有关，5.1.0的版本中DLedgerCommitLog缺少实现方法。
https://github.com/apache/rocketmq/pull/5879

**Reply by kejiancai** [reply]
嗯，我也是怀疑可能是版本的问题，支持master-slave集群，目前是发现dledger集群的定时有问题
